test/src/cookies.spec.ts

```diff
@@ -584,5 +584,139 @@ describe('Cookie specs', () => {
+      const COOKIE_NAME = 'some_cookie_name';
+
+      await page.goto(server.EMPTY_PAGE);
+      // Set a cookie for the current page.
+      await page.setCookie({
+        name: COOKIE_NAME,
+        value: 'local page cookie value',
+      });
+      expect(await page.cookies()).toHaveLength(1);
+
+      // Set a cookie for different domain.
+      await page.setCookie({
+        url: COOKIE_DESTINATION_URL,
+        name: COOKIE_NAME,
+        value: 'COOKIE_DESTINATION_URL cookie value',
+      });
+      expect(await page.cookies(COOKIE_DESTINATION_URL)).toHaveLength(1);
+
+      await page.deleteCookie({name: COOKIE_NAME});
+
+      // Verify the cookie is deleted for the current page.
+      expect(await page.cookies()).toHaveLength(0);
+
+      // Verify the cookie is not deleted for different domain.
+      await expectCookieEquals(await page.cookies(COOKIE_DESTINATION_URL), [
+        {
+          name: COOKIE_NAME,
+          value: 'COOKIE_DESTINATION_URL cookie value',
+          domain: 'example.com',
+          path: '/',
+          sameParty: false,
+          expires: -1,
+          size: 51,
+          httpOnly: false,
+          secure: true,
+          session: true,
+          sourceScheme: 'Secure',
+        },
+      ]);
+    });
+    it('should delete cookie for specified URL', async () => {
+      const {page, server} = await getTestState();
+      const COOKIE_DESTINATION_URL = 'https://example.com';
+      const COOKIE_NAME = 'some_cookie_name';
+
+      await page.goto(server.EMPTY_PAGE);
+      // Set a cookie for the current page.
+      await page.setCookie({
+        name: COOKIE_NAME,
+        value: 'some_cookie_value',
+      });
+      expect(await page.cookies()).toHaveLength(1);
+
+      // Set a cookie for specified URL.
+      await page.setCookie({
+        url: COOKIE_DESTINATION_URL,
+        name: COOKIE_NAME,
+        value: 'another_cookie_value',
+      });
+      expect(await page.cookies(COOKIE_DESTINATION_URL)).toHaveLength(1);
+
+      // Delete the cookie for specified URL.
+      await page.deleteCookie({
+        url: COOKIE_DESTINATION_URL,
+        name: COOKIE_NAME,
+      });
+
+      // Verify the cookie is deleted for specified URL.
+      expect(await page.cookies(COOKIE_DESTINATION_URL)).toHaveLength(0);
+
+      // Verify the cookie is not deleted for the current page.
+      await expectCookieEquals(await page.cookies(), [
+        {
+          name: COOKIE_NAME,
+          value: 'some_cookie_value',
+          domain: 'localhost',
+          path: '/',
+          sameParty: false,
+          expires: -1,
+          size: 33,
+          httpOnly: false,
+          secure: false,
+          session: true,
+          sourceScheme: 'NonSecure',
+        },
+      ]);
+    });
+    it('should delete cookie for specified URL regardless of the current page', async () => {
+      // Tentative test case. Depending on the cookie partitioning implementation, this
```

Comment:
This is an edge case where firefox and chrome can have observable different behavior. In ff all the cookies are partitioned, while in chrome they are not. In ff if context partition is used, delete command scoped to the cookies set by the current page. In chrome it is scoped only for the user context. As discussed offline, this is an expected to have different behavior for different implementations.